### ISSUE TITLE
Object-Oriented refactor

### ISSUE THREAD

**Issue by smangham** [andresgur/mind_the_gaps#20]
The branch `issue-oo-refactor` contains a sketch of refactoring the code to be object-oriented, following the outline diagram here:
![proposed_oo_refactor drawio](https://github.com/andresgur/mind_the_gaps/assets/8093436/6e98e01d-a4f6-4b75-8a8b-bdeebda3a30f)
I've done two of the types of `ModelComponent` (`Lorentzian` and `DampedRandomWalk`), and built the `Model` class that would collect the components.

The diagram, and an example "How I think you'd use the OO code" Python script, are in `docs/`.

**Comment by andresgur** [comment]
@smangham could you have a look at the code when you get a chance and see if we can close this one? Cheers!

**Comment by smangham** [comment]
Right, I thought I'd try to automatically generate an inheritance diagram, except that meant setting up Sphinx, which I've done...
Except of course most of that diagram is association not inheritance so I'm going to try and do it manually.

Thoughts from that so far: `Simulator.py:E13_sim` calls `simulate_lightcurve`, which as far as I can tell doesn't exist? But the tests all pass...

**Comment by andresgur** [comment]
Thanks a lot for looking into this.
E13_sim is ancient code I need to get rid of it... can you ignore it for now? There's a bunch of methods there you'll see are not used anywhere.

You are right that's a typo, it should be 
`self._lightcurve.y`

Can fix it from my side or wait for your PR?

**Comment by smangham** [comment]
Yeah, I've ignored it, it works fine otherwise. I might wait for the PR, as I've edited a fair few lines adding type hints, even if I haven't changed the actually *code*.

I think the only critique I have is that there's a real profusion of 'Model' objects and types, which is a little confusing. You have the `mean_models` which are `celerite.modeling.Model`s, `celerite_models` which are `celerite.terms.Term`s, and `psd_models` which are `astropy.modelling.Model`s. Then each different type of Model is used for a different purpose and as far as I can tell isn't exchangeable? And this is complicated by the fact you have multiple models with the same names, e.g. `psd_models.Lorentzian` and `celerite_models.Lorentzian` which makes imports pretty confusing.

I'd probably standardise the model naming, and probably include their application in them. (Disclaimer: I'm rushing and might be misremembering) E.g. renaming `Lorenztian` in `celerite_models.py` to `FittingLorentzian`, to make it more distinct from `psd_models.py`, where you renamed it `SimulatedLorentzian`?
